kartothek/utils/optimize.py

```diff
@@ -0,0 +1,13 @@
+from typing import Any
+
+import dask
+
+
+def optimize(graph: Any) -> Any:
+    """Optimize a dask graph using the dask.dataframe optimizer.
+
+    Kartothek may return only a delayed object which has no default optimizers
+    registered with dask. Therefore, we'll call the optimizer with the
+    dask.dataframe optimizer ourselves.
+    """
+    return dask.optimize(graph, optimizations=[dask.dataframe.optimize])[0]
```

Comment:
Not sure if this should be part of the public ktk API. If so, the function should have a more specific name. I'm rather thinking about calling this internally on our objects before we return, optionally toggle-able. This would then remove the need to expose this

Reply:
Removed this file from Ktk. and updated the Return description.. That we can apply `dask.optimize ` if required.

Reply:
I directly added `dask.optimize(ddf2, optimizations=[dask.dataframe.optimize])[0]` to be returned , after `repartitioning the ddf` in `repartition_ddf()` function.  Please verify if it is fine.